bootstrapper/initproto/init.proto

```diff
@@ -8,27 +8,31 @@ service API {
   rpc Init(InitRequest) returns (stream InitResponse);
 }
 
+// InitRequest is the rpc message sent to the Constellation bootstrapper to initiate the cluster bootstrapping.
 message InitRequest {
-  // repeated string autoscaling_node_groups = 1; removed
-  // bytes master_secret = 2; removed
-  string kms_uri = 3;
-  string storage_uri = 4;
-  // string key_encryption_key_id = 5; removed
-  // bool use_existing_kek = 6; removed
-  string cloud_service_account_uri = 7;
-  string kubernetes_version = 8;
-  // repeated SSHUserKey ssh_user_keys = 9; removed
-  // bytes salt = 10; removed
-  // bytes helm_deployments = 11; removed
-  // repeated uint32 enforced_pcrs = 12; removed
-  // bool enforce_idkeydigest = 13; removed
-  bool conformance_mode = 14;
-  repeated KubernetesComponent kubernetes_components = 15;
-  bytes init_secret = 16;
-  string cluster_name = 17;
-  repeated string apiserver_cert_sans = 18;
+  // KmsUri is an URI encoding access to the KMS service or master secret.
+  string kms_uri = 1;
+  // StorageUri is an URI encoding access to the storage service.
+  string storage_uri = 2;
+  // MeasurementSalt is a salt used to generate the clusterID for the initial bootstrapping node.
+  bytes measurement_salt = 3;
+  // CloudServiceAccountUri is an URI encoding the cloud service account.
+  string cloud_service_account_uri = 4;
+  // KubernetesVersion is the version of Kubernetes to install.
+  string kubernetes_version = 5;
+  // ConformanceMode is a flag to indicate whether the cluster should be bootstrapped for Kubernetes conformance testing.
+  bool conformance_mode = 6;
+  // KubernetesComponents is a list of Kubernetes components to install.
+  repeated KubernetesComponent kubernetes_components = 7;
+  // InitSecret is a secret used to authenticate the initial bootstrapping node.
+  bytes init_secret = 8;
+  // ClusterName is the name of the cluster.
+  string cluster_name = 9;
+  // ApiserverCertSans is a list of Subject Alternative Names to add to the apiserver certificate.
+  repeated string apiserver_cert_sans = 10;
```

Comment:
Idk why GH forces me to do a review rn. I just wanted to leave a comment...

I assume you are doing this to make incompatible init requests fail? A while back we decided that the numbers should not be reused. As far as I can tell, the [reasons for this](https://protobuf.dev/programming-guides/proto2/#consequences) are still valid. 
To achieve the same we could just increase the numbers to 19..29.

Reply:
I'd also like to clean this up eventually, but I also see that the style guide doesn't welcome such a change. 
A compromise could be that we remove the fields but don't re-assign their numbers? I guess we just need to keep track of the highest number we've assigned, i.e. we'd need to keep any fields after the last used one but can remove all the other fields.

Reply:
This was already merged, but I would like to quickly comment on this.

Our bootstrapper is not backward nor forward compatible, and also currently does not have to be.
Having our proto definitions be compatible in both ways doesn't help us, in fact at some occasions it produced hard to debug errors when initializing clusters with CLIs that were incompatible with older bootstrappers.
For that reason, and to finally have a cleaner proto definition I decided to remove the deprecated fields and added some comments to it.